numpy/core/src/multiarray/ctors.c

```diff
@@ -478,6 +483,18 @@ setArrayFromSequence(PyArrayObject *a, PyObject *s,
         goto fail;
     }
 
+    tmp = _array_from_array_like(s, /*dtype*/NULL, /*writeable*/0, /*context*/NULL);
```

Comment:
Should this appear above the `if` on line 480? Does it make any difference?

Reply:
I think the check on line 480 should precede both array and array-like cases. I'm not sure why it is done after `PyArray_Check`.

Reply:
I think this is fine here. Both `a` and `s` need to be verified, the order is not all that important. Note that `dim` is not used in the branch in 458-478, so the check should not be before those lines.